### ISSUE TITLE
跑tipc流程发生数据集读取错误

### ISSUE THREAD

**Issue by RuohengMa** [PaddlePaddle/PaddleOCR#11184]
- 系统环境/System Environment：ubuntu 18.04
- 版本号/Version：Paddle：develop  PaddleOCR： dygraph 问题相关组件/Related components：
- 运行指令/Command Code：
python3.7 tools/train.py -c test_tipc/configs/rec_resnet_rfl/rec_resnet_rfl.yml -o Global.use_xpu=True  Global.save_model_dir=./test_tipc/output/rec_resnet_rfl/lite_train_lite_infer/norm_train_gpus_0_autocast_null Global.epoch_num=2    Train.loader.batch_size_per_card=16
- 完整报错/Complete Error Message：
[2023/11/02 03:17:07] ppocr INFO: Initialize indexs of datasets:['./train_data/ic15_data/rec_gt_train.txt']
list index out of range
[2023/11/02 03:17:07] ppocr WARNING: The character_dict_path is None, model can only recognize number and lower letters
[2023/11/02 03:17:07] ppocr INFO: Initialize indexs of datasets:['./train_data/ic15_data/rec_gt_test.txt']
[2023/11/02 03:17:07] ppocr WARNING: The character_dict_path is None, model can only recognize number and lower letters
W1102 03:17:07.944658 104195 xpu_context.cc:151] Please NOTE: xpu device: 1
[2023/11/02 03:17:11] ppocr INFO: train dataloader has 31 iters
[2023/11/02 03:17:11] ppocr INFO: valid dataloader has 1 iters
[2023/11/02 03:17:11] ppocr INFO: train from scratch
[2023/11/02 03:17:11] ppocr INFO: During the training process, after the 0th iteration, an evaluation is run every 5000 iterations
[2023/11/02 03:17:23] ppocr INFO: epoch: [1/2], global_step: 1, lr: 0.001000, cnt_loss: 6.506174, seq_loss: 3.736380, loss: 10.242554, avg_reader_cost: 0.05737 s, avg_batch_cost: 11.70809 s, avg_samples: 16.0, ips: 1.36658 samples/s, eta: 0:11:54
[2023/11/02 03:17:35] ppocr INFO: epoch: [1/2], global_step: 2, lr: 0.001000, cnt_loss: 37.669377, seq_loss: 3.512567, loss: 41.181946, avg_reader_cost: 0.00031 s, avg_batch_cost: 11.91699 s, avg_samples: 16.0, ips: 1.34262 samples/s, eta: 0:11:48
[2023/11/02 03:17:46] ppocr INFO: epoch: [1/2], global_step: 3, lr: 0.001000, cnt_loss: 10.988602, seq_loss: 3.288755, loss: 13.682382, avg_reader_cost: 0.00041 s, avg_batch_cost: 11.45116 s, avg_samples: 16.0, ips: 1.39724 samples/s, eta: 0:11:29
[2023/11/02 03:17:58] ppocr INFO: epoch: [1/2], global_step: 4, lr: 0.001000, cnt_loss: 11.729043, seq_loss: 2.991267, loss: 14.167955, avg_reader_cost: 0.00024 s, avg_batch_cost: 11.54033 s, avg_samples: 16.0, ips: 1.38644 samples/s, eta: 0:11:15
[2023/11/02 03:18:09] ppocr INFO: epoch: [1/2], global_step: 5, lr: 0.001000, cnt_loss: 10.988602, seq_loss: 2.693780, loss: 13.682382, avg_reader_cost: 0.00030 s, avg_batch_cost: 11.50451 s, avg_samples: 16.0, ips: 1.39076 samples/s, eta: 0:11:02
[2023/11/02 03:18:21] ppocr INFO: epoch: [1/2], global_step: 6, lr: 0.001000, cnt_loss: 9.337535, seq_loss: 2.438912, loss: 11.962468, avg_reader_cost: 0.00031 s, avg_batch_cost: 11.50637 s, avg_samples: 16.0, ips: 1.39053 samples/s, eta: 0:10:49
[2023/11/02 03:18:32] ppocr INFO: epoch: [1/2], global_step: 7, lr: 0.001000, cnt_loss: 7.686469, seq_loss: 2.184045, loss: 10.242554, avg_reader_cost: 0.00029 s, avg_batch_cost: 11.50887 s, avg_samples: 16.0, ips: 1.39023 samples/s, eta: 0:10:37
[2023/11/02 03:18:44] ppocr INFO: epoch: [1/2], global_step: 8, lr: 0.001000, cnt_loss: 7.413578, seq_loss: 1.881643, loss: 9.692314, avg_reader_cost: 0.00023 s, avg_batch_cost: 11.46640 s, avg_samples: 16.0, ips: 1.39538 samples/s, eta: 0:10:25
[2023/11/02 03:18:55] ppocr INFO: epoch: [1/2], global_step: 9, lr: 0.001000, cnt_loss: 7.140687, seq_loss: 1.579242, loss: 9.142075, avg_reader_cost: 0.00031 s, avg_batch_cost: 11.37618 s, avg_samples: 16.0, ips: 1.40645 samples/s, eta: 0:10:12
[2023/11/02 03:19:07] ppocr INFO: epoch: [1/2], global_step: 10, lr: 0.001000, cnt_loss: 6.823430, seq_loss: 1.547731, loss: 8.899491, avg_reader_cost: 0.00016 s, avg_batch_cost: 11.38828 s, avg_samples: 16.0, ips: 1.40495 samples/s, eta: 0:09:59
[2023/11/02 03:19:19] ppocr INFO: epoch: [1/2], global_step: 11, lr: 0.001000, cnt_loss: 6.506174, seq_loss: 1.516220, loss: 8.656907, avg_reader_cost: 0.00020 s, avg_batch_cost: 11.84658 s, avg_samples: 16.0, ips: 1.35060 samples/s, eta: 0:09:49
[2023/11/02 03:19:30] ppocr INFO: epoch: [1/2], global_step: 12, lr: 0.001000, cnt_loss: 5.398295, seq_loss: 1.485913, loss: 7.108949, avg_reader_cost: 0.00029 s, avg_batch_cost: 11.44944 s, avg_samples: 16.0, ips: 1.39745 samples/s, eta: 0:09:37
[2023/11/02 03:19:41] ppocr INFO: epoch: [1/2], global_step: 13, lr: 0.001000, cnt_loss: 4.290416, seq_loss: 1.455606, loss: 5.560991, avg_reader_cost: 0.00019 s, avg_batch_cost: 11.38082 s, avg_samples: 16.0, ips: 1.40587 samples/s, eta: 0:09:25
[2023/11/02 03:19:53] ppocr INFO: epoch: [1/2], global_step: 14, lr: 0.001000, cnt_loss: 3.587987, seq_loss: 1.412841, loss: 4.888188, avg_reader_cost: 0.00022 s, avg_batch_cost: 11.77894 s, avg_samples: 16.0, ips: 1.35836 samples/s, eta: 0:09:14
[2023/11/02 03:20:05] ppocr INFO: epoch: [1/2], global_step: 15, lr: 0.001000, cnt_loss: 2.885558, seq_loss: 1.370077, loss: 4.215385, avg_reader_cost: 0.00029 s, avg_batch_cost: 11.35034 s, avg_samples: 16.0, ips: 1.40965 samples/s, eta: 0:09:02
[2023/11/02 03:20:16] ppocr INFO: epoch: [1/2], global_step: 16, lr: 0.001000, cnt_loss: 2.493394, seq_loss: 1.349952, loss: 3.899091, avg_reader_cost: 0.00020 s, avg_batch_cost: 11.47806 s, avg_samples: 16.0, ips: 1.39396 samples/s, eta: 0:08:50
[2023/11/02 03:20:27] ppocr INFO: epoch: [1/2], global_step: 17, lr: 0.001000, cnt_loss: 2.101231, seq_loss: 1.329827, loss: 3.582798, avg_reader_cost: 0.00015 s, avg_batch_cost: 11.37833 s, avg_samples: 16.0, ips: 1.40618 samples/s, eta: 0:08:38
[2023/11/02 03:20:39] ppocr INFO: epoch: [1/2], global_step: 18, lr: 0.001000, cnt_loss: 2.052394, seq_loss: 1.300201, loss: 3.376889, avg_reader_cost: 0.00015 s, avg_batch_cost: 11.81772 s, avg_samples: 16.0, ips: 1.35390 samples/s, eta: 0:08:28
[2023/11/02 03:20:51] ppocr INFO: epoch: [1/2], global_step: 19, lr: 0.001000, cnt_loss: 2.003556, seq_loss: 1.270575, loss: 3.170981, avg_reader_cost: 0.00020 s, avg_batch_cost: 11.38124 s, avg_samples: 16.0, ips: 1.40582 samples/s, eta: 0:08:16
[2023/11/02 03:21:02] ppocr INFO: epoch: [1/2], global_step: 20, lr: 0.001000, cnt_loss: 1.878864, seq_loss: 1.219699, loss: 3.046988, avg_reader_cost: 0.00013 s, avg_batch_cost: 11.86182 s, avg_samples: 16.0, ips: 1.34887 samples/s, eta: 0:08:05

我们提供了AceIssueSolver来帮助你解答问题，你是否想要它来解答(请填写yes/no)?/We provide AceIssueSolver to solve issues, do you want it? (Please write yes/no):

请尽量不要包含图片在问题中/Please try to not include the image in the issue.
